#!/usr/bin/env python
#

PACKAGE='racecar_potential_field_controller'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("force_scale_x",   double_t, 0, "Force scale x",   0.07,  -1, 1)
gen.add("force_scale_y",   double_t, 0, "Force scale y",   0.07,  -1, 1)
gen.add("force_offset_x",  double_t, 0, "Force offset x",  100.0, -300, 300)
gen.add("force_offset_y",  double_t, 0, "Force offset y",  0.0,   -1, 1)
gen.add("speed_p_gain",    double_t, 0, "Speed P gain",    0.05,  -1, 1)
gen.add("steering_p_gain", double_t, 0, "Steering P gain", 1.0,   -5, 5)
gen.add("steering_d_gain", double_t, 0, "Steering D gain", 0.0,   -1, 1)
gen.add("viz_forces_scale", double_t, 0, "Visualization scale factor for forces",  0.07,   -1, 1)
gen.add("viz_net_force_scale", double_t, 0, "Visualization scale factor for net force",  0.014,   -1, 1)

exit(gen.generate(PACKAGE, "potential_field_controller", "RacecarPotentialFieldController"))
